omnibus/files/private-chef-ctl-commands/chef-server-ctl

```diff
@@ -0,0 +1,328 @@
+        exec_list << "REVOKE \"#{running_package_config[key]['sql_user']}\" FROM \"#{superuser}\""
+        exec_list << "REVOKE \"#{running_package_config[key]['sql_ro_user']}\" FROM \"#{superuser}\""
+        exec_list << "DROP ROLE \"#{running_package_config[key]['sql_user']}\""
+        exec_list << "DROP ROLE \"#{running_package_config[key]['sql_ro_user']}\""
+      end
+      if perform_delete
+        delete_external_postgresql_data(exec_list, postgres)
+      else
+        path = File.join(backup_dir, "postgresql-manual-cleanup.sql")
+        dump_exec_list_to_file(path, exec_list)
+        log warn_CLEANSE002_no_postgres_delete(path, postgres)
+      end
+    end
+
+    def delete_external_postgresql_data(exec_list, postgres)
+      last = nil
+      begin
+        require "pg"
+        connection = postgresql_connection(postgres)
+        while exec_list.length > 0
+          # Preserve  our position in the list of queries to track where we left off
+          # in case of failure.
+          last = exec_list.shift
+          connection.exec(last)
+        end
+        puts " Chef Server databases and roles have been successsfully deleted from #{postgres['vip']}"
+      rescue Exception => e
+        path = nil
+        unless last.nil?
+          exec_list.insert(0, last)
+          path = Time.now.strftime("/root/%FT%R-chef-server-manual-postgresql-cleanup.sql")
+          dump_exec_list_to_file(path, exec_list)
+        end
+        # Note - we're just showing the error and not exiting, so that
+        # we don't block any other external cleanup that can happen
+        # independent of postgresql.
+        log err_CLEANSE001_postgres_failed(postgres, path, last, e.message)
+      ensure
+        connection.close if connection
+      end
+    end
+
+    def dump_exec_list_to_file(path, list)
+      File.open(path, "w") do |file|
+        list.each { |line| file.puts line }
+      end
+    end
+
+    def external_status_postgresql(detail_level)
+      postgres = external_services['postgresql']
+      begin
+        require "pg"
+        connection =postgresql_connection(postgres)
+        if detail_level == :sparse
+          # We connected, that's all we care about for sparse status
+          # We're going to keep the format similar to the existing output
+          "run: postgresql: connected OK to #{postgres['vip']}:#{postgres['port']}"
+          # to hopefully avoid breaking anyone who parses this.
+        else
+          max_conn = connection.exec("SELECT setting FROM pg_settings WHERE name = 'max_connections'")[0]['setting']
+          total_conn = connection.exec('SELECT sum(numbackends) num FROM pg_stat_database')[0]['num']
+          version = connection.exec('SHOW server_version')[0]['server_version']
+          lock_result =  connection.exec('SELECT pid FROM pg_locks WHERE NOT GRANTED')
+          locks = lock_result.map { |r| r['pid'] }.join(",")
+          if locks.nil? or locks.empty?
```

Comment:
I think our general ruby style would use || rather than `or` here.  Similarly in a few other conditionals.